### PR TITLE
Fix win32 build, add checkpoints, update block headers, update fork heights, update version

### PR DESCRIPTION
**Pull request by fluffypony**
I don't know how I ended up with 3 commits for the first thing, and I'm too tired to fiddle with git rebase right now.

### CODE REVIEW

File: src/wallet/api/wallet.cpp [hunk 1]
```diff
@@ -632,12 +632,12 @@ void WalletImpl::refreshThreadFunc()
...
 
     while (true) {
-        std::unique_lock<std::mutex> lock(m_refreshMutex);
+        boost::mutex::scoped_lock lock(m_refreshMutex);
         if (m_refreshThreadDone) {
             break;
         }
         LOG_PRINT_L3(__FUNCTION__ << ": waiting for refresh...");
-        m_refreshCV.wait_for(lock, std::chrono::seconds(m_refreshIntervalSeconds));
+        m_refreshCV.wait(lock);
```
Comment (moneromooo-monero) [inline src/wallet/api/wallet.cpp]: I recall some dodgy code here, not sure that will actually work, need pointing out to mbg033 to make sure

**Comment by moneromooo-monero** [comment]
ok except first comment

**Comment by moneromooo-monero** [comment]
ok

File: src/cryptonote_core/blockchain.cpp [hunk 2]
```diff
@@ -100,6 +106,7 @@ static const struct {
   // version 2 starts from block 624634, which is on or around the 23rd of November, 2015. Fork time finalised on 2015-11-20. No fork voting occurs for the v2 fork.
   { 2, 624634, 0, 1445355000 },
 
+  // versions 3-5 were passed in rapid succession from September 18th, 2016
```
Comment (NanoAkron) [inline src/cryptonote_core/blockchain.cpp]: This may read better with 'approved' rather than 'passed'

**Pull request merged**